### PR TITLE
chore(docs): refactoring guides and some other nits

### LINKED ISSUE #6222: Merge "Getting Started" documentation sections into one simple, end-to-end page
### Problem

The current "Getting Started" part of the Noir docs cover multiple sections, namely:
- Install Nargo
- Install Proving Backend
- Creating a Project

Image

Developers have to read through at least 3 pages just to create, prove and verify their first Noir project, which makes onboarding unnecessarily complex.

### Happy Case

Tasks:
- [ ] Merge the "Install Nargo", "Install Proving Backend" and "Creating a Project" pages → one "Getting Started" page
- [ ] Split "Alternative Installations" and "Project Breakdown" pages out into standalone pages

The set of instructions to get started end-to-end (i.e. tooling installation and creating a project) shall be covered in 1 page, where developers should feel confident understanding the basics after reading through it.

Example: https://docs.aztec.network/guides/developer_guides/getting_started

### Additional Context

Broken out from https://github.com/noir-lang/noir/pull/6175 and related discussions.

### Project Impact

Nice-to-have

### PR DESCRIPTION
**Pull request by signorecello**
This closes the remaining work for restoring some of the devex regression around noir and bb, and other docs improvements including homepage removal and redirects set-up.

- [x] Made "Getting Started" E2E with `bbup` and `bb` as examples (closes #6222 and #6223)
- [x] Removed BB specific documentation
- [x] Updated docusaurus to latest
- [x] Made homepage CTA open the docs in the same tab

### CODE REVIEW

**Review by TomAFrench — CHANGES_REQUESTED** [review]
This PR is enshrining noirenberg as the primary installation method for Noir [which isn't what was agreed on slack](https://aztecprotocol.slack.com/archives/C06RUPLLKPA/p1726847849000279?thread_ts=1726830818.086199&cid=C06RUPLLKPA) and is a "backtrace into the valley" as @Savio-Sou put it. It can be placed in "Alternative installations" under noirup or under `bb` in "Proving backend installation" but it's not an installation method which is supported and tested by the Noir team so it should not take priority over noirup.

We should also not merge the "hello noir" page into the installation tooling page as they're two separate things and if we're having multiple installation methods then it doesn't make sense to inline that content into just one of them.

**Comment by critesjosh** [comment]
do we want to remove the homepage? I thought we were going to to keep it (@Savio-Sou) since you don't typically first land on a docs site when going a web page. I know the homepage right now isn't the prettiest, but at least it works as a placeholder so if we want to update the homepage in the future we won't have to restructure again.

Removing the homepage will also break all existing links to noir docs out on the web since it removes `/docs/` from the url

**Comment by signorecello** [comment]
I was under the impression we wanted to remove it but didn't because of the redirects, but I fixed the redirects in netlify.toml (links with /docs/ redirect to the correct pages)

Glad to keep it, though, if it's useful

**Comment by Savio-Sou** [comment]
> do we want to remove the homepage?

We would like to update it in the long run.

I would prefer keeping it unless it is stirring confusions / frictions in our community.

**Comment by Savio-Sou** [comment]
Created https://github.com/noir-lang/noir/issues/6222 and https://github.com/noir-lang/noir/issues/6223 (also partly relevant https://github.com/AztecProtocol/aztec-packages/issues/8530) from this PR.

Recommend closing and splitting this PR up to tackle the Issues individually; easier to review and merge that way.

File: docs/docs/getting_started.md
Comment (Savio-Sou) [inline docs/docs/getting_started.md:74]:
```suggestion
nargo execute
```

Following https://github.com/noir-lang/noir/pull/6031, `nargo execute` now writes a witness file named after the Noir package by default.

File: docs/docs/getting_started.md
Comment (Savio-Sou) [inline docs/docs/getting_started.md:84]:
```suggestion
bb prove -b ./target/hello_world.json -w ./target/hello-world.gz -o ./target/proof
```

File: docs/.markdownlint.json
Comment (TomAFrench) [inline docs/.markdownlint.json:1]: Why is this removed?

File: docs/docs/getting_started.md
Comment (TomAFrench) [inline docs/docs/getting_started.md:11]: Who currently owns this domain?

File: docs/docs/getting_started.md
Comment (TomAFrench) [inline docs/docs/getting_started.md:29]: Note that script currently doesn't exist.

**Comment by Savio-Sou** [comment]
On my recent commit https://github.com/noir-lang/noir/pull/6175/commits/b9801d0153a209da6007962d8b97d2a7171cc246, not sure if that conflicts with any best practices, but having Quick Start as a standalone page feels more natural to me than as the Getting Started category page.

(Sorry for pushing it directly, not sure how to cleanly commit locally edits as PR suggestions; feel free to revert if undesirable.)

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [inline docs/docs/getting_started/quick_start.md:94]:
```suggestion
The proof is now generated in the `target` folder.

### 5. Verify the execution proof

To verify it we first need to compute the verification key from the compiled circuit, and use it to verify:
```

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [inline docs/docs/getting_started/quick_start.md:109]:
```suggestion
```

More appropriate to highlight in the "Private & Public Types" section of the "Data Types" page than in "Quick Start"

**Review by Savio-Sou — COMMENTED** [review]
Suggestions for the "Quick Start" page

File: docs/docs/getting_started/noir_installation.md
Comment (Savio-Sou) [inline docs/docs/getting_started/noir_installation.md:2]:
```suggestion
title: Advanced Noir Installation
```

"Standalone" doesn't seem very intuitive for what this page covers. Maybe "Advanced"?

Comment (Savio-Sou) [reply docs/docs/getting_started/noir_installation.md:2]: We could consider moving this page to another section / standalone as well

File: docs/.markdownlint.json
Comment (signorecello) [reply docs/.markdownlint.json:1]: Because I wanted to remove the only rule there so it didn't make sense to keep the file

File: docs/docs/getting_started.md
Comment (signorecello) [reply docs/docs/getting_started.md]: I have it for the time being

File: docs/docs/getting_started.md
Comment (signorecello) [reply docs/docs/getting_started.md]: Yes it is pending on the other PR

File: docs/docs/getting_started/noir_installation.md
Comment (signorecello) [reply docs/docs/getting_started/noir_installation.md:2]: We want `noirup` as the standard way to install Nargo, so I wouldn't call it "advanced". Just "Installation" would be confusing though specially as it mostly a reference.

I'll think about other options and iterate later on

Comment (Savio-Sou) [reply docs/docs/getting_started/noir_installation.md:2]: > We want noirup as the standard way to install Nargo, so I wouldn't call it "advanced".

I would consider `noirup` and `noirup -v ...` to be the common installation path, where all other `noirup ...` installation paths to be advanced usage.

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:35]: I interpret the framework as to refrain from strict steps unless there's specific guidance required (ex. tutorial too big for early and frequent results)

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:63]:
```suggestion
### Compiling and executing

We can now use `nargo` to generate a _Prover.toml_ file, where our input values will be specified:

```sh
cd hello_world
nargo check
```
```

We can assume the user knows what `cd`  accomplishes, so bundling the two commands minimizes distraction

Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:63]: We can assume the user knows what `cd` accomplishes, so bundling the two commands minimizes distraction

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:80]:
```suggestion
## Proving backend
	
Different proving backends may provide different tools and commands to work with Noir programs. Here Barretenberg's `bb` CLI tool is used as an example:
```

Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:80]: This allows us to remove the next line which is always good

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [inline docs/docs/getting_started/quick_start.md:82]:
```suggestion

```

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:109]: I made it an admonition as to try to avoid the distraction, but I'd rather have a specific how-to guide on this than bury it in the concepts section. 

This is one of the most asked questions ever since nargo stopped proving and outputting the proof in hex format, so I'll leave it as visible as possible for the time being

Comment (Savio-Sou) [reply docs/docs/getting_started/quick_start.md:111]: Feel free to reference this Issue if helpful: https://github.com/AztecProtocol/aztec-packages/issues/6487

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [reply docs/docs/getting_started/quick_start.md:36]: > framework
As in Diataxis?

> unless there's specific guidance required
Not sure if I follow, aren't we specifying specific steps and commands that user should run

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [inline docs/docs/getting_started/quick_start.md:8]:
```suggestion

### Noir

```

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [inline docs/docs/getting_started/quick_start.md:17]:
```suggestion

### Proving backend

```

**Review by Savio-Sou — COMMENTED** [review]
Supplement minor sectioning under _Installation_ for better readability

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [inline docs/docs/getting_started/quick_start.md:76]:
```suggestion
nargo execute
```

Following https://github.com/noir-lang/noir/pull/6031, `nargo execute` now writes a witness file named after the Noir package by default

File: docs/docs/getting_started/quick_start.md
Comment (Savio-Sou) [reply docs/docs/getting_started/quick_start.md:64]: @signorecello FYI looks like this wasn't committed (in case you wanted to)

**Review by Savio-Sou — APPROVED** [review]
I still think removing the "Steps 1-6" framing from the current docs makes the E2E flow more obscure than easier to follow

File: docs/docs/getting_started.md
Comment (Savio-Sou) [reply docs/docs/getting_started.md]: To minimize complications / risks, best to:

- First transfer ownership, or
- Revert to the GitHub hyperlink
 
before merging.

Comment (TomAFrench) [reply docs/docs/getting_started.md]: Agreed, I'd like this to be an "officially owned" domain before we push people towards it.

File: docs/docs/getting_started.md
Comment (TomAFrench) [reply docs/docs/getting_started.md]: Yes, I'm flagging this as we shouldn't merge this change until that's merged.

File: docs/.markdownlint.json
Comment (TomAFrench) [reply docs/.markdownlint.json:1]: I mean, why remove the rule?

File: docs/docs/getting_started.md
Comment (signorecello) [reply docs/docs/getting_started.md]: I'll transfer the ownership before merging so we don't have to iterate later

File: docs/docs/getting_started/quick_start.md
Comment (signorecello) [reply docs/docs/getting_started/quick_start.md:64]: Yeah seems like it doesn't commit if I "resolve" the conversation before. Just noticed I didn't merge any of your previous changes 😒

TIL i guess

File: docs/docs/getting_started.md
Comment (TomAFrench) [reply docs/docs/getting_started.md]: This domain is still pointing at https://raw.githubusercontent.com/AztecProtocol/aztec-packages/refs/heads/zpedro/bbup_and_docs/barretenberg/bbup/install which now 404s

Comment (Savio-Sou) [reply docs/docs/getting_started.md]: Both `noirup` and `bbup` seems to work

File: docs/.markdownlint.json
Comment (signorecello) [reply docs/.markdownlint.json:1]: It was erroring somehow, can't remember exactly, thought I could just remove it

File: docs/docs/getting_started/quick_start.md
Comment (TomAFrench) [inline docs/docs/getting_started/quick_start.md:28]:
```suggestion
You can use the `bbup` installation script to quickly install and update BB, Barretenberg's CLI tool:
```

**Review by TomAFrench — APPROVED** [review]
Looks good to me now URLs are transferred.

**Pull request merged**
